examples/lm1b/train.py

```diff
@@ -365,6 +452,23 @@ def main(argv):
           eval_summary_writer.scalar(key, val, step)
         eval_summary_writer.flush()
 
+      # Fast inference of prompt extension using trained LM.
+      rng, subrng = jax.random.split(rng)
+      pred_rngs = random.split(subrng, jax.local_device_count())
+      prompt = jnp.array(encoder.encode(FLAGS.prompt))
+      prompt = jax_utils.replicate(prompt)
+      prompt = jnp.reshape(prompt, (prompt.shape[0], 1, prompt.shape[1]))
+      cache = jax_utils.replicate(
+          cache_def.initialize_cache((1, FLAGS.max_predict_token_length)))
```

Comment:
initialize_cache creates zero constants. So it might be faster / more memory efficient when initlaize_check is inside predict_step this way they will not be materialized as device buffers

Reply:
We'd still need to provide the cache_def as a static arg though, no?  The amount of memory used here is tiny right now though.